### PR TITLE
feat: Add S3 rename tool

### PR DESCRIPTION
**Pull request by Alima777**
This tool is used to delete table_uuid from the key of s3 objects. Only patterns like "root_prefix/table_uuid/part_uuid/key" will be renamed, other objects will keep like before.

### CODE REVIEW

File: programs/part-toolkit/RenameS3Object.cpp [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    S3::Model::CopyObjectRequest request;
+
+    request.WithCopySource(fromBucket + "/" + fromKey)
+            .WithKey(toKey)
+            .WithBucket(toBucket);
+
+    S3::Model::CopyObjectOutcome outcome = client.CopyObject(request);
+    if (!outcome.IsSuccess()) {
+        const S3::S3Error &err = outcome.GetError();
+        std::cerr << "Error: CopyObject: " <<
```
Comment (linkwk7) [inline programs/part-toolkit/RenameS3Object.cpp:27]: 日志的输出都改成用Poco::Logger吧

File: programs/part-toolkit/RenameS3Object.cpp [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        std::cerr << "Error: ListObjects: " <<
+                  outcome.GetError().GetMessage() << std::endl;
+        return false;
+    }
+    else {
+        Vector<S3::Model::Object> objects = outcome.GetResult().GetContents();
+        boost::asio::thread_pool thread_pool(threadNum);
+        for (S3::Model::Object &object: objects) {
+            // root_prefix/table_uuid/part_uuid/name
+            String from_key = object.GetKey();
```
Comment (linkwk7) [inline programs/part-toolkit/RenameS3Object.cpp:78]: The key will have format of prefix/table_uuid/part_uuid/data, can check if table_uuid & part_uuid is a valid uuid here by ReaderHelpers.h:tryReadUUIDText

File: programs/part-toolkit/RenameS3Object.cpp [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+bool CopyObject(const String &fromBucket, const String &fromKey, const String &toBucket, 
+                    const String &toKey, const S3::S3Client & client) {
+    S3::Model::CopyObjectRequest request;
+
+    request.WithCopySource(fromBucket + "/" + fromKey)
+            .WithKey(toKey)
+            .WithBucket(toBucket);
+
+    S3::Model::CopyObjectOutcome outcome = client.CopyObject(request);
```
Comment (linkwk7) [inline programs/part-toolkit/RenameS3Object.cpp:24]: This api seems only support object with less than 5GB, UploadPartCopy should allow copy part larger than 5GB

Comment (Alima777) [reply programs/part-toolkit/RenameS3Object.cpp:39]: Fixed.

**Comment by hustnn** [comment]
hi @Alima777 , pls sign the CLA

**Comment by Alima777** [comment]
Hi, @hustnn , it's signed now.

**Review by linkwk7 — APPROVED** [review]

**Pull request merged**